www/js/diary/addressNamesHelper.ts

```diff
@@ -139,8 +156,16 @@ async function fetchNominatimLocName(loc_geojson) {
 
 // Schedules nominatim fetches for the start and end locations of a trip
 export function fillLocationNamesOfTrip(trip: CompositeTrip) {
-  nominatimLimiter.schedule(() => fetchNominatimLocName(trip.end_loc));
-  nominatimLimiter.schedule(() => fetchNominatimLocName(trip.start_loc));
+  [trip.start_confirmed_place, trip.end_confirmed_place].forEach((place) => {
+    if (place?.geocoded_address || place?.reverse_geocode) {
+      const coordsStr = place.location.coordinates.toString();
+      const data = place.reverse_geocode || { address: place.geocoded_address };
+      localStorage.setItem(coordsStr, JSON.stringify(data));
+      publish(coordsStr, JSON.stringify(data));
```

Comment:
nice! great idea to reuse the data from prior mappings even for trips that don't have the address instead of looking it up.